tests/unit/test_attribute.py

```diff
@@ -66,12 +71,44 @@ def test_tags(run, store):
     store.log.assert_called_with(tags_remove={"sys/tags": ("tag1", "tag2")})
 
 
-def test_series(run, store):
+def test_append(run, store):
     run["sys/series"].append(1, step=1, timestamp=10)
-    store.log.assert_called_with(metrics={"sys/series": 1}, step=1, timestamp=10)
+    store.log.assert_called_with(metrics=Metrics(data={"sys/series": 1}, step=1), timestamp=10)
 
     run["sys/series"].append({"foo": 1, "bar": 2}, step=2)
-    store.log.assert_called_with(metrics={"sys/series/foo": 1, "sys/series/bar": 2}, step=2, timestamp=None)
+    store.log.assert_called_with(metrics=Metrics(data={"sys/series/foo": 1, "sys/series/bar": 2}, step=2), timestamp=None)
+
+    run["my/series"].append({"foo": 1, "bar": 2}, step=3, preview=True, preview_completion=0.3)
+    store.log.assert_called_with(
+            metrics=Metrics(data={"my/series/foo": 1, "my/series/bar": 2}, step=3, preview=True, preview_completion=0.3),
+            timestamp=None)
+
+
+@freeze_time("2024-07-30 12:12:12.000022")
+def test_extend(run, store):
+    now = datetime.now()
+    before = now - timedelta(seconds=1)
+
+    run["my/series"].extend([1,2], steps=[1,2], timestamps=[before, now])
```

Comment:
I'd feel a bit better if step and value were not equal - if they are, we're not testing if they're not swapped.